url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/server/ClientExample.java

```diff
@@ -0,0 +1,302 @@
+
+
+            SSLContext ctx = SSLContext.getInstance("SSLv3");
+            ctx.init(null, null, null);
+
+
+            String absoluteKeyPath = new File(Thread.currentThread().getContextClassLoader().getResource(ConnectionConfiguration.DEFAULT_KEY_STORE).toURI()).getAbsolutePath();
+            System.out.println(absoluteKeyPath);
+            System.setProperty("javax.net.ssl.keyStore", absoluteKeyPath);
+            System.setProperty("javax.net.ssl.keyStorePassword", ConnectionConfiguration.DEFAULT_KEY_STORE_PASSWORD);
+            System.setProperty("java.protocol.handler.pkgs", "com.sun.net.ssl.internal.www.protocol");
+
+            String absolutePath = new File(Thread.currentThread().getContextClassLoader().getResource("pingpong.crt").toURI()).getAbsolutePath();
+            System.out.println(absolutePath);
+            System.setProperty("javax.net.ssl.trustStore", absoluteKeyPath);
+            System.setProperty("javax.net.ssl.trustStorePassword", "pingpong123");
+
+            // SSLSocketFactory factory = (SSLSocketFactory) ctx.getSocketFactory();
+            SSLSocketFactory factory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            // SSLSocket client = (SSLSocket) factory.createSocket("localhost", 5011);
+            Socket client = (SSLSocket) factory.createSocket("localhost", 5011);
+            //  client.startHandshake();
+            // java.security.cert.Certificate[] serverCerts = client.getSession().getPeerCertificates();
+            ObjectInputStream inputStream = new ObjectInputStream(client.getInputStream());
+            int readInt = inputStream.readInt();
+            System.out.println("SSL- I was assigned Client id: " + readInt);
+            ObjectOutputStream outputStream = new ObjectOutputStream(client.getOutputStream());
+            outputStream.writeObject(new byte[]
+                    {
+                        1, 3, 2, 3
+                    });
+            outputStream.flush();
+            outputStream.close();
+        }
+    }
+
+    public static void ssl2() throws IOException,
+                                     InterruptedException,
+                                     NoSuchAlgorithmException,
+                                     KeyManagementException,
+                                     KeyStoreException,
+                                     UnrecoverableKeyException,
+                                     CertificateException,
+                                     URISyntaxException
+    {
+        {
+
+
+
+            // SSLSocketFactory factory = HttpsURLConnection.getDefaultSSLSocketFactory();
+
+
+            KeyStore ks = KeyStore.getInstance("JKS");
+            String absolutePath = new File(Thread.currentThread().getContextClassLoader().getResource(ConnectionConfiguration.DEFAULT_KEY_STORE).toURI()).getAbsolutePath();
+            ks.load(new FileInputStream(absolutePath), (ConnectionConfiguration.DEFAULT_KEY_STORE_PASSWORD).toCharArray());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)